turbinia/pubsub.py

```diff
@@ -94,6 +99,130 @@ def from_json(self, json_str):
     self.__dict__ = obj
 
 
+class TurbiniaCelery(object):
+  """Celery app object.
+
+  Attributes:
+    app (Celery): The Celery app itself.
+    _fexec: Lets us initialize the Celery task after configuring the app, since
+        the Celery 'app.task' decorator is unknown before initialization.
+    fexec (function, args, kwargs): Celery task, which can run functions
+        without needing to pre-register them (similar to PSQ). Takes in a
+        function name, as well as any arguments (named or otherwise). All
+        workers must have this function defined.
+  """
+
+  def __init__(self):
+    """Celery configurations."""
+    self.app = None
+    self.fexec = None
+
+  def _fexec(self):
+    """Closure used to pass functions to workers."""
+    @self.app.task(name='fexec')
+    def fexec(f, *args, **kwargs):
+      """Lets us pass in an arbitrary function without Celery annotations"""
+      return f(*args, **kwargs)
+    return fexec
+
+  def setup(self):
+    """Set up Celery"""
+    config.LoadConfig()
+    self.app = celery.Celery(
+        'turbinia',
+        broker=config.CELERY_BROKER,
+        backend=config.CELERY_BACKEND
+    )
+    self.app.conf.update(
+        task_default_queue=config.INSTANCE_ID,
+        event_serializer='pickle',
```

Comment:
Seems you can do some Task class inheritance things: http://docs.celeryproject.org/en/latest/whatsnew-4.0.html#the-task-base-class-no-longer-automatically-register-tasks

So something like this might work `@app.task(bind=True, base=TurbiniaTask)`, but I will need to do more testing / figure that out (in a future PR probably).